client/trino-client/src/main/java/io/trino/client/OkHttpUtil.java

```diff
@@ -226,7 +226,7 @@ public static void setupSsl(
             sslContext.init(keyManagers, new TrustManager[] {trustManager}, null);
 
             clientBuilder.sslSocketFactory(sslContext.getSocketFactory(), trustManager);
-            clientBuilder.hostnameVerifier(LegacyHostnameVerifier.INSTANCE);
+            clientBuilder.hostnameVerifier(OkHostnameVerifier.INSTANCE);
```

Comment:
This is a behavioral change. The legacy verifier allows certificates which use the CN rather than SubjectAltName. While this violates modern TLS specifications, it is common in internal environments, and even our own unit tests (unless we've since fixed that).

Reply:
I've added a legacy hostname verifier with changes required for it to work (some of the code that this class is depending on was removed)